apps/app/components/Preview/index.tsx

```diff
@@ -263,6 +263,62 @@ export const PreviewComponent: React.FC = () => {
     )
   }, [userProps, detail])
 
+  const mailAddress: string = useMemo(
+    () => userProps?.defaultAddress ?? 'unknown',
+    [userProps]
+  )
+
+  const toMessage = useMemo(() => {
+    const isOfficeMail = OFFICE_ADDRESS_LIST.some(
+      (address) => detail?.from.address === address
+    )
+
+    if (isOfficeMail && detail && detail.to === null) {
+      return [
+        {
+          address: mailAddress,
+        },
+      ]
+    }
+
+    return detail?.to || []
+  }, [detail])
+
+  const avatarList = useMemo(() => {
+    if (!detail) return []
+    const exists: Array<string> = []
+    let arr = [detail.from, ...toMessage]
+    if (detail.cc) arr = [...arr, ...detail.cc]
+    if (detail.bcc) arr = [...arr, ...detail.bcc]
+
+    arr = arr.filter(({ address }) => {
+      if (exists.includes(address.toLocaleLowerCase())) return false
+      exists.push(address.toLocaleLowerCase())
+      return true
+    })
+    return arr
+  }, [detail, toMessage])
+
+  const replyAllList = useMemo(() => {
+    if (!detail) return []
+    const exists: Array<string> = [
```

Comment:
You can also use [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) here.